uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/leave-request-model_test.js

```diff
@@ -53,12 +54,36 @@ define([
       });
     });
 
-    it('balanceChangeByAbsenceType() calls equivalent API method', function () {
-      var leaveRequestPromise = LeaveRequest.balanceChangeByAbsenceType();
+    describe('balanceChangeByAbsenceType()', function () {
 
-      leaveRequestPromise.then(function () {
-        expect(LeaveRequestAPI.balanceChangeByAbsenceType).toHaveBeenCalled();
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        var leaveRequestPromise = LeaveRequest.balanceChangeByAbsenceType();
```

Comment:
Also, this is not representative of how you use the method. You would never call the method without parameters in the real code, so why are you doing it in the test?